sql/src/main/java/org/apache/druid/sql/calcite/rule/DruidRules.java

```diff
@@ -223,7 +225,14 @@ public void onMatch(final RelOptRuleCall call)
                              .withSort(sort)
         );
         if (outerQueryRel.isValidDruidQuery()) {
-          call.transformTo(outerQueryRel);
+          try {
+            call.transformTo(outerQueryRel);
+          }
```

Comment:
Would be super-helpful to include comments on code such as this. Calcite is hard enough to understand already: every hint helps.

This occurs in the sort rule: `Sort <-- DruidRel` --> `DruidRel(with sort)`. Before, if the replacement failed, we'd just throw the exception. Here, we're catching an exception and ignoring the rewrite.

This seems a bit fishy. First, That sort has to be pushed into the query, right? We have no way to express a sort as an "operator" without it being part of a native query.

Second, waiting for it to fail, then ignoring the error, seems...odd. Should we not check whatever the condition is we want to ignore, and avoid doing the rewrite in that case? But, that raises the above question: if we don't rewrite away that sort here, when will we do it?

Point is: Calcite is really complex. Sometimes the fix is not where you first find the problem, it can be multiple steps back where we introduced a fault.

Reply:
You're right to be suspicious. This change definitely isn't right. This try/catch is papering over a bug in some implementation of DruidRel. Any DruidRel that returns `true` from `isValidDruidQuery()` should never throw a `CannotBuildQueryException` from either of its `toDruidQueryForExplaining` or `toDruidQuery` methods. Instead of doing the try/catch here, the buggy DruidRel implementation should be fixed.

Reply:
Thanks for your comments. I'll investigate further

Reply:
Removed the try/catch thing as this is getting fixed upstream